packages/cdktf/lib/terraform-variable.ts

```diff
@@ -154,7 +154,7 @@ export class TerraformVariable
       type: this.type,
       sensitive: this.sensitive,
       nullable: this.nullable,
-      validation: this.validation.map((validation) => ({
+      validation: this?.validation.map((validation) => ({
```

Comment:
This would check whether `this` is undefined. See below for checking `this.validation`:
```suggestion
      validation: this.validation?.map((validation) => ({
```